File: flutter/lib/models/order_request.dart
class OrderRequest {
  final int id;
  final String reference;
  final String supplier;
  final String warehouse;
  final String driver;
  final int nbOrders;
  final int nbPackages;
  final String createdAt;
  final String status;

  OrderRequest({
    required this.id,
    required this.reference,
    required this.supplier,
    required this.warehouse,
    required this.driver,
    required this.nbOrders,
    required this.nbPackages,
    required this.createdAt,
    required this.status,
  });

  factory OrderRequest.fromJson(Map<String, dynamic> json) {
    return OrderRequest(
      id: json['id'] ?? 0,
      reference: json['reference'] ?? '',
      supplier: json['supplier'] ?? '',
      warehouse: json['warehouse'] ?? '',
      driver: json['driver'] ?? '',
      nbOrders: json['nb_orders'] ?? 0,
      nbPackages: json['nb_packages'] ?? 0,
      createdAt: json['created_at'] ?? '',
      status: json['status'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'reference': reference,
      'supplier': supplier,
      'warehouse': warehouse,
      'driver': driver,
      'nb_orders': nbOrders,
      'nb_packages': nbPackages,
      'created_at': createdAt,
      'status': status,
    };
  }
}

File: delivery.itlbservices.com/orders/models/order_request.py
from django.db import models

from users.models import User, Supplier, Employee
from core.models import Warehouse
from .packages import DeliveryPriceList

class OrderRequestStatuses(models.TextChoices):
    requested = "requested", "Requested"
    confirmed = "confirmed", "Confirmed"
    cancelled = "cancelled", "Cancelled"
    picked_up = "picked_up", "Picked Up"
    
class OrderRequest(models.Model):
    reference = models.CharField(max_length=50)
    supplier = models.ForeignKey(Supplier, on_delete=models.SET_NULL, null=True)
    driver = models.ForeignKey(Employee, on_delete=models.SET_NULL, null=True)
    nb_orders = models.IntegerField()
    nb_packages = models.IntegerField()
    total_amount = models.FloatField(default=0)
    status = models.CharField(choices=OrderRequestStatuses.choices, max_length=20, default=OrderRequestStatuses.requested)
    created_by = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name="request_created_by")
    cancelled_by = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name="request_cancelled_by")
    created_at = models.DateTimeField()
    warehouse = models.ForeignKey(Warehouse, on_delete=models.SET_NULL, null=True)
    delivery_pricelist = models.ForeignKey(DeliveryPriceList, on_delete=models.SET_NULL, null=True)
    
    def __str__(self):
        return self.reference